client/testing/block_deliverer_mock.go

```diff
@@ -0,0 +1,169 @@
+package testing
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/msp"
+	"github.com/pkg/errors"
+	"github.com/s7techlab/hlf-sdk-go/v2/api"
+)
+
+type blocksReader struct {
+	//  <channel-name> => [<block1.pb>,...<blockN.pb>]
+	data             map[string][]*common.Block
+	closeWhenAllRead bool
+	closer           func() error
+}
+
+// NewBlocksDelivererMock - read all blocks in proto format from provided folder.
+// returns sdkapi.BlocksDeliverer - interface
+// closeWhenAllRead - close channel when all data have been written
+// closerFunc - close func that will be returned from Blocks()
+func NewBlocksDelivererMock(rootPath string, closeWhenAllRead bool, closerFunc func() error) (api.BlocksDeliverer, error) {
```

Comment:
Accept Interfaces Return Struct

Reply:
and why closerFunc in arguments ?

Reply:
there are no reasons for making `blocksReader` struct public

Reply:
and no obstacles too